connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/SparkConnectClientSuite.scala

```diff
@@ -482,6 +485,72 @@ class SparkConnectClientSuite extends ConnectFunSuite with BeforeAndAfterEach {
     iter.foreach(_ => ())
     assert(reattachableIter.resultComplete)
   }
+
+  test("GRPC stub unary call throws error immediately") {
```

Comment:
Tests added in this file are checking assumptions under which retrying errors operates.